### ISSUE TITLE
LuaStack::read to std::string fails

### ISSUE THREAD

**Issue by aszkid** [AdUki/LuaState#8]
I've been using this library for a few days and i'm enjoying it so far. However, i've found one problem that appears when compiling with GCC 4.7.2 (haven't been able to test on other compilers, sorry):

``` c++
std::string hello = state["mystr"]
```

It outputs this error when compiling:

```
In function `std::string lua::Value::to<std::string>() const':
LuaValue.h:185: undefined reference to `std::string lua::stack::read<std::string>(lua_State*, int)'
```

So i've just added this to LuaStack.h, right after the `const char*` specialization:

``` c++
template<>
inline std::string read(lua_State* luaState, int index) {
        return lua_tostring(luaState, index);
}
```

And now it works without any apparent problem. I suspect this bug is caused by my compiler, which is not able to transform the `const char*` specialization to an `std::string`.

I have not used any function from the library except for simple value retrievals, so i don't know if this bug has furthest reach; just in case i open this issue to address those problems if they occur.

Thanks for your work, i really enjoy working with this library!

**Comment by AdUki** [comment]
This is not working on clang I think, but i can merge it for now. I read that implicit conversions always make problems. I am thinking about slightly extend API that it will be possible setting and returning type without much boilerplate code...

**Comment by aszkid** [comment]
I've compiled a demo using this fix on Clang++ 3.5.0 (trunk 205115) in a 64bit machine, and it works without problems. Clang may or may not use this function, but it does not break it.

**Comment by AdUki** [comment]
Now i tried it. It works for your code but not for this:

``` cpp
std::string hello;
hello = state["mystr"];
```

You will get:

```
use of overloaded operator '=' is ambiguous (with operand types 'std::string' (aka 'basic_string<char, char_traits<char>, allocator<char> >') and 'lua::Value')
    hello = state["mystr"];
    ~~~~~ ^ ~~~~~~~~~~~~~~
```

Because C++ don't know whether use assignment operator or implicit constructor...

I added some functions so mostly you will specify type:

``` cpp
std::string hello;
hello = state["mystr"].toString();
```
